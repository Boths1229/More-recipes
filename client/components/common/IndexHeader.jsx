import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description functional component to render the landing page Header
 *
 * @returns {JSX} return JSX
 */
const IndexHeader = () => (
  <header>
    <nav className="navbar navbar-expand-lg navbar-light bg-green">
      <Link to="#" className="navbar-link nav-text" id="navlink">More-recipe</Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item">
            <Link to="/recipes" className="navbar-link nav-text" id="navlink">Recipes</Link>
          </li>
        </ul>
        <form className="form-inline my-2 my-lg-0">
          <input
            className="form-control mr-sm-2"
            type="search"
            placeholder="Search"
            aria-label="Search"
          />
          <button className="btn search-btn my-2 my-sm-0" type="submit">Search</button>
        </form>
      </div>
    </nav>
  </header>
);

export default IndexHeader;
