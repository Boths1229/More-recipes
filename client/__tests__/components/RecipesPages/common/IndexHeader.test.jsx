import React from 'react';
import { shallow } from 'enzyme';
import IndexHeader from '../../../../components/common/IndexHeader';

describe('IndexHeader component', () => {
  it('matches snapshot', () => {
    const output = shallow(<IndexHeader />);
    expect(output).toMatchSnapshot();
  });
});
