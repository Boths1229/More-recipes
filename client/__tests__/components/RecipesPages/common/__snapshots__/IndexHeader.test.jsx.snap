// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexHeader component matches snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-green"
  >
    <Link
      className="navbar-link nav-text"
      id="navlink"
      replace={false}
      to="#"
    >
      More-recipe
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="navbar-link nav-text"
            id="navlink"
            replace={false}
            to="/recipes"
          >
            Recipes
          </Link>
        </li>
      </ul>
      <form
        className="form-inline my-2 my-lg-0"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="search"
        />
        <button
          className="btn search-btn my-2 my-sm-0"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </nav>
</header>
`;
